From 728ae9df017d348b4cc85c5c65ce7f88d80c9c6c Mon Sep 17 00:00:00 2001
From: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
Date: Fri, 29 May 2026 14:05:43 +0200
Subject: [PATCH 24/29] mtd: spi-nor: Fix MEMUNLOCK for IS25LP256

Restore software lock support for IS25LP256 by setting
SPI_NOR_HAS_LOCK and SPI_NOR_4BIT_BP.

Without this, the MTD unlock callback is not registered and userspace
MEMUNLOCK fails with -EOPNOTSUPP.

Signed-off-by: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
---
 drivers/mtd/spi-nor/issi.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mtd/spi-nor/issi.c b/drivers/mtd/spi-nor/issi.c
index ed7ead8bb7b18..080e628bef17f 100644
--- a/drivers/mtd/spi-nor/issi.c
+++ b/drivers/mtd/spi-nor/issi.c
@@ -46,7 +46,8 @@ static const struct flash_info issi_parts[] = {
 			     SECT_4K | SPI_NOR_DUAL_READ) },
 	{ "is25lp256",  INFO(0x9d6019, 0, 64 * 1024, 512,
 			     SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ |
-			     SPI_NOR_4B_OPCODES)
+			     SPI_NOR_HAS_LOCK | SPI_NOR_4B_OPCODES |
+			     SPI_NOR_4BIT_BP)
 		.fixups = &is25lp256_fixups },
 	{ "is25lp512",  INFO(0x9d601a, 0, 64 * 1024, 1024,
 			     SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ |
-- 
2.44.4

